### PR TITLE
tests/console: Refactor LoginToFedora into LoginToGeneric

### PR DESCRIPTION
**Pull request by 0xFelix**
**What this PR does / why we need it**:

This change refactors LoginToFedora so it can be used as a generic
template for different VMs. This allows to test logins on guests which
work similar to Fedora (e.g. CentOS or CoreOS/RHCOS).

Needed by https://github.com/kubevirt/containerdisks/pull/18

**Release note**:

```release-note
None
```

### CODE REVIEW

File: tests/console/login.go [hunk 1]
```diff
@@ -19,6 +19,24 @@ import (
 // LoginToFunction represents any of the LoginTo* functions
 type LoginToFunction func(*v1.VirtualMachineInstance) error
 
+// LoginOptions contains values needed when logging in to a generic VM
+type LoginOptions struct {
```
Comment (EdDev) [inline tests/console/login.go:23]: I think you can try and make the current nicer by using an object:
```
func LoginToFedora(vmi *v1.VirtualMachineInstance) error {
    c := NewConsole("myname", "mypassword", vmi.Name)
    return c.Login()
}
```

**Review by EdDev — COMMENTED** [review]
See inline an alternative suggestion.

**Comment by 0xFelix** [comment]
Closing this for now, it is no longer needed by https://github.com/kubevirt/containerdisks/pull/18.

**Pull request closed without merging**